awx/api/views/__init__.py

```diff
@@ -370,6 +370,16 @@ class InstanceDetail(RetrieveUpdateAPIView):
     model = models.Instance
     serializer_class = serializers.InstanceSerializer
 
+    def update(self, request, *args, **kwargs):
+        r = super(InstanceDetail, self).update(request, *args, **kwargs)
+        if status.is_success(r.status_code):
+            obj = self.get_object()
+            if obj.capacity != 0:
+                obj.set_capacity_value()
+                obj.save()
```

Comment:
@matburt the prior state of code here is that we called `obj.refresh_capacity()` when someone adjusted the capacity adjustment slider. That method would actually run `psutil.virtual_memory().total` on the node the request was made to. Realistically, all nodes probably have the same memory/CPU specs for ordinary clusters. Moving forward with execution nodes, however, I'd expect them to be very different machines.

So now, the capacity adjustment changes the `capacity` field and that's it. I just realized I should add `update_fields` to make that super clear.

Reply:
👍🏻 that works for me.